### DISCUSSION TITLE
Contract Verification

### DISCUSSION THREAD

**Discussion by AztecWarlord** [Cyfrin/foundry-full-course-cu#57] [Q&A] [answered]
I am receiving this issue when trying to deploy and verify my FundMe contract on the Sepolia network:
```terminal
ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.00166435500776699 ETH (554785 gas * avg 3.000000014 gwei)

We haven't found any matching bytecode for the following contracts: [0x715c9f51ac108ea9dfc5f0fe52cb57f1ac7a6aee].

This may occur when resuming a verification, but the underlying source code or compiler version has changed.
##
Start verification for (0) contracts
All (0) contracts were verified!
```
Any assistance to fix this issue and get the Contract verified with the correct bytecode would be greatly appreciated.

**Comment by alymurtazamemon** [comment]
@AztecWarlord Are you doing it manually?

**Reply by AztecWarlord** [reply]
I am trying to do it from the terminal with the makefile

**Reply by alymurtazamemon** [reply]
@AztecWarlord Ok, So can you show your command and the contract you are verifying?

**Reply by alymurtazamemon** [reply]
@AztecWarlord If it is resolved then please give some details about your solution here for others so anyone facing this issue can see the solution here. Thanks

And then mark that solution as answer.

**Reply by AztecWarlord** [reply]
sorry I am new to this I reopend the issue following Patrick's guild lines in a new issue in general discussions not realizing I could have moved this to general I will tag you in the new discussion. Also, the issue has not been resolved

**Reply by AztecWarlord** [reply]
@alymurtazamemon the contract address is:
```
0x715c9f51ac108ea9dfc5f0fe52cb57f1ac7a6aee
```
the command is:
```
make deploy ARGS="--network sepolia"
```
Thank you for your assistance.

**Reply by alymurtazamemon** [reply]
@AztecWarlord Which lesson is it? I will see the maker file of that lesson.

**Reply by AztecWarlord** [reply]
@alymurtazamemon Lession 7: Foundry FundMe.

**Reply by alymurtazamemon** [reply]
@AztecWarlord Do you have `ETHERSCAN_API_KEY` in your `.env` file?

For me it is working fine;
```bash
alymurtazamemon@Alis-MacBook-Air foundry-fund-me-f23 % make deploy ARGS="--network sepolia"
[⠆] Compiling...
No files changed, compilation skipped
Traces:
  [1082350] DeployFundMe::run() 
    ├─ [505212] → new HelperConfig@0xC7f2Cf4845C6db0e1a1e91ED41Bcd0FcC1b0E141
    │   └─ ← 2191 bytes of code
    ├─ [381] HelperConfig::activeNetworkConfig() [staticcall]
    │   └─ ← 0x694AA1769357215DE4FAC081bf1f309aDC325306
    ├─ [0] VM::startBroadcast() 
    │   └─ ← ()
    ├─ [486534] → new FundMe@0x1777889731A65C526f26E5FF3e07405164D7A81C
    │   └─ ← 2318 bytes of code
    ├─ [0] VM::stopBroadcast() 
    │   └─ ← ()
    └─ ← FundMe: [0x1777889731A65C526f26E5FF3e07405164D7A81C], HelperConfig: [0xC7f2Cf4845C6db0e1a1e91ED41Bcd0FcC1b0E141]


Script ran successfully.

== Return ==
0: contract FundMe 0x1777889731A65C526f26E5FF3e07405164D7A81C
1: contract HelperConfig 0xC7f2Cf4845C6db0e1a1e91ED41Bcd0FcC1b0E141

## Setting up (1) EVMs.
==========================
Simulated On-chain Traces:

  [576722] → new FundMe@0x1777889731A65C526f26E5FF3e07405164D7A81C
    └─ ← 2318 bytes of code


==========================

Chain 11155111

Estimated gas price: 3.000001194 gwei

Estimated total gas used for script: 749738

Estimated amount required: 0.002249214895187172 ETH

==========================

###
Finding wallets for all the necessary addresses...
##
Sending transactions [0 - 0].
⠁ [00:00:00] [###########################################################] 1/1 txes (0.0s)
Transactions saved to: /Users/alymurtazamemon/Desktop/Blockchain/patrick-repos/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/11155111/run-latest.json

##
Waiting for receipts.
⠉ [00:00:25] [#######################################################] 1/1 receipts (0.0s)
##### sepolia
✅ Hash: 0x69900a11f52a6df183e26d3523fa10701a53f63a88cb0110da785f271961638d
Contract Address: 0x1777889731a65c526f26e5ff3e07405164d7a81c
Block: 3681539
Paid: 0.001730646339206616 ETH (576882 gas * 3.000000588 gwei)


Transactions saved to: /Users/alymurtazamemon/Desktop/Blockchain/patrick-repos/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/11155111/run-latest.json



==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.001730646339206616 ETH (576882 gas * avg 3.000000588 gwei)
##
Start verification for (1) contracts
Start verifying contract `0x1777889731a65c526f26e5ff3e07405164d7a81c` deployed on sepolia

Submitting verification for [src/FundMe.sol:FundMe] "0x1777889731A65C526f26E5FF3e07405164D7A81C".

Submitting verification for [src/FundMe.sol:FundMe] "0x1777889731A65C526f26E5FF3e07405164D7A81C".

Submitting verification for [src/FundMe.sol:FundMe] "0x1777889731A65C526f26E5FF3e07405164D7A81C".

Submitting verification for [src/FundMe.sol:FundMe] "0x1777889731A65C526f26E5FF3e07405164D7A81C".
Submitted contract for verification:
        Response: `OK`
        GUID: `ctgcxv9gbmpnmuzhtrtxea8vzppiytaldyeynwsqrzfkzwlkzc`
        URL:
        https://sepolia.etherscan.io/address/0x1777889731a65c526f26e5ff3e07405164d7a81c
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
Contract verification status:
Response: `OK`
Details: `Pass - Verified`
Contract successfully verified
All (1) contracts were verified!

Transactions saved to: /Users/alymurtazamemon/Desktop/Blockchain/patrick-repos/foundry-fund-me-f23/broadcast/DeployFundMe.s.sol/11155111/run-latest.json

alymurtazamemon@Alis-MacBook-Air foundry-fund-me-f23 % 
```

**Reply by alymurtazamemon** [reply]
@AztecWarlord This is another way you can verify the existing contracts;
```bash
forge verify-contract --chain-id 11155111 --constructor-args $(cast abi-encode "constructor(address)" price-feed-address-here) --etherscan-api-key <your_etherscan_api_key> <the_contract_address> src/FundMe.sol:FundMe
```
Try this.

**Reply by AztecWarlord** [reply]
@alymurtazamemon Yes, I do have `ETHERSCAN_API_KEY` my `.env` file.

**Reply by AztecWarlord** [reply]
> @AztecWarlord This is another way you can verify the existing contracts;
> 
> ```shell
> forge verify-contract --chain-id 11155111 --constructor-args $(cast abi-encode "constructor(address)" price-feed-address-here) --etherscan-api-key <your_etherscan_api_key> <the_contract_address> src/FundMe.sol:FundMe
> ```
> 
> Try this.

I will try this.

**Reply by alymurtazamemon** [reply]
@AztecWarlord Try these, if it does not work then check your etherscan key or recreate a new one.

**Reply by AztecWarlord** [reply]
@alymurtazamemon I followed your suggestions and created a new etherscan key and still received  the same result.

**Reply by cromewar** [reply]
Trying to re create this as well but with no luck as my contract gets verified properly.

**Reply by PatrickAlphaC** [reply]
Can you run `foundryup` @AztecWarlord and try again? They had some verification issues recently.

**Reply by AztecWarlord** [reply, marked as answer]
@PatrickAlphaC I ran ```foundryup``` and ran the makefile and now the issue has been fixed and the contract is verified thank you! And thanks again @alymurtazamemon @cromewar for your insight.

**Reply by alymurtazamemon** [reply]
@AztecWarlord Mark your answer as answered so others can see this in the answered category. Thanks

**Reply by AztecWarlord** [reply]
@alymurtazamemon how do I do that?

**Reply by alymurtazamemon** [reply]
@AztecWarlord Ah, you have selected the general category for it you need to select the Q&As category, after that you will find the Mark Answer option below each comment and you can mark yours.

**Comment by PatrickAlphaC** [comment]
Hello! Thanks for making this issue. 

Can you:
1. Make this a discusson on the full repo? https://github.com/Cyfrin/foundry-full-course-f23/discussions

2. Could you please remember to follow this section to format your questions? https://youtu.be/umepbfKp5rI?t=22464
